src/DEA/allDEA.R

```diff
@@ -269,65 +235,30 @@ for (sheet in sheet_names) {
 
 #### TCGA analysis ####
 
-FCS <- list()
-log2FCS <- list()
-p_vals <- list()
-sig_FCS <- list()
-genes_sig_list <- list()
+TCGA_data_sig <- list()
+TCGA_data_non_sig <- list()
 
 wt_mean <- rowMeans(wt_TPM)
-count = 1
-
 
+count = 1
 for(CL in TCGA_mut_data) {
   
   genes = TCGA_mut_data[[count]]$symbol
-  FC = mut_TPM[ ,count] / wt_mean
-  
-  log2FC = log2(FC)
-  bool_inf_pos <- (is.infinite(log2FC) & (sign(log2FC) == 1))
-  bool_inf_neg <- (is.infinite(log2FC) & (sign(log2FC) == -1))
-  
-  genes_pos_inf <- genes[bool_inf_pos]
-  genes_neg_inf <- genes[bool_inf_neg]
-  
-  #find NaN and inf values and keep only the non-NaN and non-inf values
-  bool <- (is.nan(log2FC)|is.infinite(log2FC))
-  FC_real <- FC[!bool]
-  log2FC_real <- log2FC[!bool]
-  genes_real <- c(genes[!bool], genes_pos_inf, genes_neg_inf)
+  mut = mut_TPM[ ,count]
+  ENSG = TCGA_mut_data[[count]]$ensID
   
-  #set fold change of inf genes to the max fold change and combine with real values
-  FC_comb <- c(FC_real, rep(max(FC_real),length(genes_pos_inf)), rep(min(FC_real),length(genes_neg_inf)))
-  log2FC_comb <- c(log2FC_real, rep(max(log2FC_real),length(genes_pos_inf)), rep(min(log2FC_real),length(genes_neg_inf)))
-  FCS[[count]] <- FC_comb
-  log2FCS[[count]] <- log2FC_comb
-  
-  #need to specify mean and standard deviation
-  mean_fc <- mean(log2FC_real)
-  stddev <- sd(log2FC_real)
-  p_value <- 2*pnorm(log2FC_real, mean_fc, stddev)
-  p_value <- c(p_value, rep(10^(-50), (length(genes_pos_inf)+length(genes_neg_inf))))
-  p_vals[[count]] <- p_value
-  
-  bool_p <- (p_value <= 0.05)
-  genes_sig <- genes_real[bool_p]
-  p_values_keep <- p_vals[bool_p]
-  fold_change_sig <- FC_comb[bool_p]
-  
-  sig_FCS[[count]] <- fold_change_sig
-  genes_sig_list[[count]] <- genes_sig
-  
-  hist(log2FC_real, probability = T, breaks = 50,
-       main = count,
-       xlab = 'Log2 of Fold Change')
+  temp_data <- DEA(genes,wt_mean,mut,ENSG)
+  TCGA_data_sig[[count]] <- temp_data[[1]]
+  TCGA_data_non_sig[[count]] <- temp_data[[2]]
   
   count = count + 1
 }
 
-dev.off()
+export(TCGA_data_sig, '~jennadiegel/Documents/SURE/DES/TCGA_data/TCGA_data_sig.xlsx')
+export(TCGA_data_non_sig, '~jennadiegel/Documents/SURE/DES/TCGA_data/TCGA_data_non_sig.xlsx')
+
+
+
```

Comment:
Sweet! Glad also you reduced the line number from 331 to 261! As another best coding practice, this is something to strive for - making your code more efficient and elegant.